src/classes/Collision/CapsuleConvexHull.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * @param {Goblin.RigidBody|Goblin.RigidBodyProxy} objectA
+ * @param {Goblin.RigidBody|Goblin.RigidBodyProxy} objectB
+ * @param {boolean} doLightweightCollision
+ * @returns {Goblin.ContactDetails[]|null}
+ */
+Goblin.Collision.capsuleConvexHull = ( function() {
+    var innerSegmentStart = new Goblin.Vector3();
+    var innerSegmentEnd = new Goblin.Vector3();
+
+    return function( objectA, objectB, doLightweightCollision ) {
+        var capsule;
+        var convexHull;
+        if ( objectA.shape.shapeType === Goblin.Shapes.Type.CapsuleShape ) {
+            capsule = objectA;
+            convexHull = objectB;
+        } else {
+            capsule = objectB;
+            convexHull = objectA;
+        }
+
+        innerSegmentStart.set( 0, capsule.shape.half_height, 0 );
+        innerSegmentEnd.set( 0, -capsule.shape.half_height, 0 );
+        capsule.transform.transformVector3( innerSegmentStart );
+        capsule.transform.transformVector3( innerSegmentEnd );
+
+        // If closestPointOnHull is not null then inner segment does not collide with convex hull.
+        // This can mean two things - either there is no collision at all or there is a shallow collision.
+        var closestPointOnHull = Goblin.GjkEpa.findClosestPointOnObjectToLineSegment( convexHull, innerSegmentStart, innerSegmentEnd );
```

Comment:
Does this method use GJK/EPA at all? Or it simply sits there?

Reply:
Yes, one of the main purposes of GJK (which we didn't use before) - is to find closest points between objects